src/python_testing/matter_testing_support.py

```diff
@@ -963,43 +963,51 @@ def get_test_desc(self, test: str) -> str:
         except AttributeError:
             return test
 
-    # Sends an out-of-band command to a Matter app
-    def write_to_app_pipe(self, command):
+    def write_to_app_pipe(self, app_pipe_name: str, command_dict: dict):
```

Comment:
Why are we requesting `app_pipe_name` here. We should be able to pick this from `self.app_pip`, as is still explained in the documentation. See the example below.
```
class A():
     def __init__(self):
         self.a = "default"
     def useA(self):
         print(self.a)

class B(A):
     def setA(self, a):
         self.a = a


a = A()
a.useA() # "default"

b = B()
b.useA() # "default"

b.setA("test")
b.useA() # "test"
```

Reply:
Given that this is something that is only set once at the start, it would make sense to have `app_pipe` added to the base class. This way we do not need to pass the same thing every time this method is used.

Reply:
@tcarmelveilleux any objections to this approach?

Reply:
The issue is that sometimes you need to control 2 apps, so always need to provide app pipe name. We should have helpers that auto-selects a default. Let's do that as a follow-up.

```
def write_to_app_pipe(self, command_dict: dict, app_pipe_name: Optional[str] = None):
    if app_pipe_name is None:
        app_pipe_name = self.get_default_app_pipe_name()
```

For now we would be better off having `app_pipe_name` at the end (like above signature) so that when it becomes optional, we don't have to pass None each time.